### ISSUE TITLE
Mypy reports invalid type when a value is a parameterized generic

### ISSUE THREAD

**Issue by danpascu** [python/mypy#17532]
**Bug Report**

When I attempt to use a parameterized generic as a value to build a signature at runtime, mypy complains about the parameter as being invalid, when the parameter comes from a variable.

**To Reproduce**

```python
from inspect import Parameter

def signature_parameter(name: str, item_type: type) -> Parameter:
    return Parameter(name=name, kind=Parameter.KEYWORD_ONLY, annotation=list[item_type])
```

**Expected Behavior**

I would expect no error in this case (when the parameterized type is used as a value).

I know a variable cannot be used as a type parameter in a typing context (i.e. when used as a type annotation).
I understand that the following code would be wrong because variables cannot be used in type specifications:

```python
var: type = int
lst: list[var] = [1, 2, 3]  # this is wrong
```

But in my case above, `list[item_type]` is not used as a type specification, but as a value, that is passed to the runtime to build a signature. I do not think mypy should flag such use of values. It should simply consider the value to be of type `types.GenericAlias`.

**Actual Behavior**

```
typing-8.py:6: error: Variable "item_type" is not valid as a type  [valid-type]
typing-8.py:6: note: See https://mypy.readthedocs.io/en/stable/common_issues.html#variables-vs-type-aliases
Found 1 error in 1 file (checked 1 source file)
```

**Your Environment**

- Mypy version used: `mypy 1.12.0+dev.6a0657e5959ba1777c4d427f8f355d499035d145 (compiled: no)`
- Mypy command-line flags: None
- Mypy configuration options from `mypy.ini` (and other config files):
```
[tool.mypy]
enable_incomplete_feature = "NewGenericSyntax"
disable_bytearray_promotion = true
disable_memoryview_promotion = true
check_untyped_defs = true
warn_unreachable = true
warn_redundant_casts = true
warn_unused_ignores = true
```
- Python version used: 3.12

**Comment by ngnpope** [comment]
I think you want to be using `typing.TypeAlias` instead of `type`.

See https://mypy-play.net/?mypy=latest&python=3.12&gist=6880667b46e349905950555df8dfaa91

I'm not sure whether using `TypeAlias` to annotate a function parameter is valid - it doesn't seem to complain... - and I expect that you're probably waiting on `TypeExpr` from [PEP 747](https://peps.python.org/pep-0747/).

**Comment by danpascu** [comment]
Note: I checked the code and TypeAlias would not even work. I simplified the example for brevity, but in the real code `item_type` comes from an instance attribute of a generic class that is annotated with:

```python
from inspect import Parameter

class ListFieldDescriptor[T]:
    name: str
    item_type: type[T]

    @property
    def signature_parameter(self) -> Parameter:
        return Parameter(name=self.name, kind=Parameter.KEYWORD_ONLY, annotation=list[self.item_type])
```

I cannot replace `type[T]` with `TypeAlias[T]` for multiple reasons:
 1. it's used by other parts of the code that expects `type[T]`
 2. mypy doesn't even accept `TypeAlias[T]` as valid there without a right hand side value which I don't have yet
 3. I get new errors (see [playground example](https://mypy-play.net/?mypy=master&python=3.12&gist=feda89a31f5aa78082e7a72615659ac5&enable-incomplete-feature=NewGenericSyntax)) that I reported in #17531
